### ISSUE TITLE
Unable to Migrate Products

### ISSUE THREAD

**Issue by chrissm79** [ubertheme/magento2_data_migration#4]
I'm getting the following error message on Step 5 (when attempting to migrate products).  Is this an issue with my data or with the migration? Thanks!

``` php
CDbCommand failed to execute the SQL statement: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '16309-117-0' for key 'CATALOG_PRODUCT_ENTITY_VARCHAR_ENTITY_ID_ATTRIBUTE_ID_STORE_ID'. The SQL statement executed was: INSERT INTO `catalog_product_entity_varchar` (`attribute_id`, `store_id`, `entity_id`, `value`) VALUES (:yp0, :yp1, :yp2, :yp3)
```

**Comment by quynhvv** [comment]
Hi chrissm79,
What are the Magento1 and Magento2 versions in your working?
What version of this tool you are using?
Please follow full steps of this tool at https://github.com/ubertheme/magento2_data_migration
Regards,
quynhvv

**Comment by chrissm79** [comment]
Hi @quynhvv 
It looks like it was an issue with my own data being dirty. I had to check for a match (`Mage2CatalogProductEntityVarchar` with the same `entity_id`, `store_id` and `attribute_id`). If I found a match I just skipped it. It slowed down the migration (due to my additional check), but everything migrated over successfully!

On a side note, I'm working with a store that has over 18k sales and 180k quotes, so I couldn't get through step 7 without getting an error about running out of memory. I currently moved the code for migrating sales_order and quote_item into a while loop that handles 1000 rows at a time. Still in the process of the migration but it seems to be working.

**Comment by quynhvv** [comment]
Yes, let's contact me if you need further help.
Regards,
Mall.

On Tue, Oct 20, 2015 at 9:21 PM, chrissm79 notifications@github.com wrote:

> Closed #4 https://github.com/ubertheme/magento2_data_migration/issues/4.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ubertheme/magento2_data_migration/issues/4#event-440387655
> .

## 

Best regards, Quynh Vu Van.

## http://www.magentocommerce.com/certification/directory/dev/1639183/

Skype:  quynh.vv
Mobile:  098 525 2002
